lib/commands/context.js

```diff
@@ -73,7 +80,121 @@ extensions.getNewRemoteDebugger = async function getNewRemoteDebugger () {
     garbageCollectOnExecute: util.hasValue(this.opts.safariGarbageCollect)
       ? !!this.opts.safariGarbageCollect
       : false,
-  });
+    udid: this.opts.udid,
+  }, this.isRealDevice());
+};
+
+commands.setContext = async function setContext (name, callback, skipReadyCheck) {
+  function alreadyInContext (desired, current) {
+    return (desired === current ||
+           (desired === null && current === NATIVE_WIN) ||
+           (desired === NATIVE_WIN && current === null));
+  }
+
+  log.debug(`Attempting to set context to '${name}'`);
+  if (alreadyInContext(name, this.curContext)) {
+    // already in the named context, no need to do anything
+    return;
+  } else if (name === NATIVE_WIN || name === null) {
+    // switching into the native context
+    this.curContext = null;
+    return;
+  }
+
+  // switching into a webview context
+
+  // if contexts have not already been retrieved, get them
+  if (_.isUndefined(this.contexts)) {
+    await this.getContexts();
+  }
+
+  let contextId = name.replace(WEBVIEW_BASE, '');
+  if (contextId === '') {
+    // allow user to pass in "WEBVIEW" without an index
+    // the second context will be the first webview as
+    // the first is always NATIVE_APP
+    contextId = this.contexts[1];
```

Comment:
should we check if context's length suffice first?

Reply:
No need. This line will not error, and the user will get what we want (i.e., a no such context error) on the next check.

Reply:
if the first element does not exist then the value of `contextId` will be `undefined`, however we only compare this.curContext with `null` below

Reply:
On line 119-120 (i.e., immediately after this), there is the check
```js
if (!_.includes(this.contexts, contextId)) {
  throw new errors.NoSuchContextError();
}
```

Reply:
good, just wanted to make sure it cannot be equal to undefined